### PR TITLE
Hw02 task02

### PR DESCRIPTION
**Pull request by julwein**

### CODE REVIEW

File: homework_02/task_02.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+"""
+from collections import Counter
+from typing import List, Tuple
+
+
+def major_and_minor_elem(inp: List[int]) -> Tuple[int, int]:
+    """
+    Return tuple (most_common_element, least_common_element).
+    """
+    el_occurance = Counter(inp).most_common()
```
Comment (Cjkjvfnby) [inline homework_02/task_02.py]: What is the complexity of this line (Hint: look into source code)?

Could we do better?

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]
Solution works, tests are ok, need to refactor code a bit.

File: homework_02/task_02.py (hunk 1 above)
Comment (julwein) [reply homework_02/task_02.py]: AFAICT, we have here creating a dictionary and then sorting it by value, which gives an average complexity of O(n * n * log n) = O(n^2 * log n). Sure enough, sorting is a bit overkill when we need just max and min. So, I see how I can reduce the complexity to O(n^2). Is that it?

Comment (Cjkjvfnby) [reply homework_02/task_02.py]: `dictionary and then sorting it`  create and sort is done one after another so you need to add them O(n) + O(n * log n) -> O(n * log n).

> So, I see how I can reduce the complexity to O(n^2). Is that it?

Almost.  You need O(n)

**Review by Cjkjvfnby — APPROVED** [review]

**Pull request merged**